components/collection/drop/MintStepper.vue

```diff
@@ -4,29 +4,29 @@
     v-model="amountToMint"
     data-testid="drop-stepper-container"
     :max="max"
+    :disabled="!isLogIn"
     class="[&>.neo-input]:h-full [&>.neo-input>input]:h-full md:w-[200px]" />
 </template>
 <script lang="ts" setup>
 import { NeoStepper } from '@kodadot1/brick'
 import useHolderOfCollection from '@/composables/drop/useHolderOfCollection'
 
+const { isLogIn } = useAuth()
 const { availableNfts } = useHolderOfCollection()
-const { amountToMint, drop } = storeToRefs(useDropStore())
+const { amountToMint, drop, mintsCount } = storeToRefs(useDropStore())
 
 const show = computed(() => drop.value.type !== 'free')
 
 const max = computed(() => {
-  const holderMax =
-    drop.value.type === 'holder'
-      ? availableNfts.serialNumbers.length
-      : undefined
-  // tmp remove when uploading to IPFS step can be skipped @see https://github.com/kodadot/nft-gallery/issues/10001#issuecomment-2041533819
-  const dropMax = DROP_MASSMINT_LIMIT[drop.value.alias] ?? undefined
+  const options = [
+    drop.value.max && drop.value.max - mintsCount.value,
+    DROP_MASSMINT_LIMIT[drop.value.alias] ?? undefined, // tmp remove when uploading to IPFS step can be skipped @see https://github.com/kodadot/nft-gallery/issues/10001#issuecomment-2041533819
+  ]
 
-  if (holderMax) {
-    return dropMax ? Math.min(dropMax, holderMax) : holderMax
+  if (drop.value.type === 'holder') {
+    options.push(availableNfts.serialNumbers.length)
   }
 
-  return dropMax
+  return Math.min(...options.filter((option) => option !== undefined))
```

Comment:
Options returns at worst case:

false
Undefined
And if it is a holder of drop it will return a number too

+ should not be it a max?